pkg/kubelet/client/kubelet_client.go

```diff
@@ -93,7 +93,8 @@ func (c *KubeletClientConfig) transportConfig() *transport.Config {
 			KeyFile:  c.KeyFile,
 			KeyData:  c.KeyData,
 		},
-		BearerToken: c.BearerToken,
+		BearerToken:       c.BearerToken,
+		ConnectionTimeout: 7 * time.Second,
```

Comment:
There was no old default – this is a new setting introduced as part of this PR. 7 seconds is a somewhat arbitrary choice but works well in my tests and feels like a decent balance of being a "long time" (in computer terms) as well as being short enough to re-establish a connection before other timeouts in the controller manager kick in to mark the Kubelet as dead.